docs/editor-neovim.qmd

```diff
@@ -0,0 +1,63 @@
+---
+title: "Neovim"
+editor:
+  markdown:
+    wrap: sentence
+    canonical: true
+---
+
+# Installing the air client
+
+First, install the Air client by following the instructions on the [Air CLI documentation](https://posit-dev.github.io/air/cli.html).
+
+# Configuration
+
+Air can be configured as a Language Server Protocol (LSP) client via [nvim-lspconfig](https://github.com/neovim/nvim-lspconfig) or as a formatter plugin via [conform.nvim](https://github.com/stevearc/conform.nvim).
+
+## LSP Configuration with nvim-lspconfig
+
+To configure Air as an LSP client, add the following to your `nvim/lua/plugins/lspconfig.lua`:
+
+```lua
+require("lspconfig").air.setup({})
+```
+
+To enable automatic formatting on save, add the following to your `nvim/after/plugin/lsp.lua` (or other appropriate location):
+
+```lua
+vim.api.nvim_create_autocmd("BufWritePre", {
+    pattern = { "*.R", "*.r" },
+    callback = function()
+        vim.lsp.buf.format({
+            async = true,
+            filter = function(client)
+                return client.name == "air"
+            end,
+        })
+    end,
+})
+```
+
+## conform.nvim configuration
+
+Air can be configured as a formatter plugin via [conform.nvim](https://github.com/stevearc/conform.nvim). This allows you to format R code with Air inside code chunks in Quarto and RMarkdown documents. It can be configured by adding the following to your `nvim/lua/plugins/conform.lua`:
```

Comment:
```suggestion
Air can be used as a formatter via [conform.nvim](https://github.com/stevearc/conform.nvim). This allows using Air to format R code inside code chunks in Quarto and RMarkdown documents. It can be configured by adding the following to your configuration:
```

> This allows you to format R code with Air inside code chunks in Quarto and RMarkdown documents.

Is this an out-of-the-box functionality or does one need to configure tree-sitter injections? 👀

Reply:
I think it is out of the box, my conform config is pretty simple.

Reply:
@TymekDev and @PMassicotte can you please double check that Air works in various Quarto and RMarkdown documents? I have my doubts because I went looking in conform and could not find any specific Quarto support for LSP/formatting passthrough.

If it does seem to be working, I'd love to know what I missed so I can study how it works on the conform side

Reply:
> @TymekDev and @PMassicotte can you please double check that Air works in various Quarto and RMarkdown documents? I have my doubts because I went looking in conform and could not find any specific Quarto support for LSP/formatting passthrough.
> 
> If it does seem to be working, I'd love to know what I missed so I can study how it works on the conform side

@DavisVaughan conform.nvim supports injected language formatting this probably what is being meant here, the see the following: https://github.com/stevearc/conform.nvim/blob/master/doc/advanced_topics.md#injected-language-formatting-code-blocks